tests/test_core/test_events.py

```diff
@@ -353,3 +353,116 @@ def test_rotate(
 
     data_minimal_rotated.rotate(0)
     assert data_minimal.events.equals(data_minimal_rotated.events)
+
+
+@pytest.mark.unit
+def test_slice_new_object(
+    example_events_data_minimal,
+    example_events_data_minimal_none,
+    example_events_data_frameclock,
+) -> None:
+
+    # Arrange
+    data_minimal = Events(example_events_data_minimal)
+    data_none = Events(example_events_data_minimal_none)
+    data_frameclock = Events(example_events_data_frameclock)
+
+    # Act
+    data_minimal_full = data_minimal.slice(inplace=False)
+    data_minimal_full_arguments = data_minimal.slice(start=1.1, end=2.3, inplace=False)
+    data_minimal_end_sliced = data_minimal.slice(end=2.2, inplace=False)
+    data_minimal_start_sliced = data_minimal.slice(start=1.2, inplace=False)
+    data_none_sliced = data_none.slice(inplace=False)
+    data_frameclock_sliced = data_frameclock.slice(
+        end=15, slice_by="frameclock", inplace=False
+    )
+
+    # Assert
+    assert data_minimal_full.events.shape == (2, 2)
+    assert data_minimal_full_arguments.events.shape == (2, 2)
+    assert data_minimal_end_sliced.events.shape == (1, 2)
+    assert data_minimal_start_sliced.events.shape == (1, 2)
+    assert data_none_sliced.events.shape == (1, 2)
+    assert data_frameclock_sliced.events.shape == (1, 3)
```

Comment:
to me it feels a bit unstable to only check the DF shape. could you incorporate assert statements testing if the actual columns match?